### ISSUE TITLE
Password Hash in SignUp/Login

### ISSUE THREAD

**Issue by Javi510348** [uprm-inso4101-2023-2024-s2/semester-project-rumaradoptionapp#41]
We need to find a way of hashing the password during signup, so we can prevent storing the password in the database as plain-text.

Due Date: 02/19/2024

**Comment by evnd04** [comment]
Agreed, there is probably many examples on the internet you could find.

**Comment by Javi510348** [comment]
A hashing algorithm has been selected for the passwords and it's called "argon2".  It's one of the newest hashing passwords algorithms in the market.  It's considered a key derivation function and it was selected as the winner in password hashing competition in 2015.

To hash the password with argon2, we need to use the function .hash("your_password") to make a hash of our password. Once the hash is created, the generated hash is used as the password for the database.  In the situation that a user wants to login to their user account, he should to use it's original password without issues because argon2 provides the capability of verifying if the password is valid or not.  The function .verify("hash","original_password") will check if this is the original password that was used. 

If you have any questions, please let me know.

**LINKS**
https://www.npmjs.com/package/@node-rs/argon2/v/1.0.4

https://www.cryptolux.org/images/0/0d/Argon2.pdf

https://argon2-cffi.readthedocs.io/en/stable/argon2.html
